scrapy/core/downloader/handlers/aiohttp.py

```diff
@@ -0,0 +1,102 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+import aiohttp
+
+from scrapy import responsetypes
+from scrapy.http.headers import Headers
+from scrapy.utils.defer import deferred_from_coro
+from scrapy.utils.reactor import (
+    is_asyncio_reactor_installed,
+    set_asyncio_event_loop,
+    verify_installed_reactor,
+)
+
+if TYPE_CHECKING:
+    from typing import Self
+
+    from twisted.internet.defer import Deferred
+
+    from scrapy.crawler import Crawler
+    from scrapy.http.request import Request
+    from scrapy.http.response import Response
+    from scrapy.settings import Settings
+    from scrapy.spiders import Spider
+
+
+class AiohttpDownloadHandler:
+    def __init__(self, settings: Settings, crawler: Crawler):
+        if not is_asyncio_reactor_installed():
+            raise ValueError(
+                "AiohttpDownloadHandler requires the asyncio Twisted "
+                "reactor. Make sure you have it configured in the "
+                "TWISTED_REACTOR setting. See the asyncio documentation "
+                "of Scrapy for more information."
+            )
+
+        verify_installed_reactor(
+            "twisted.internet.asyncioreactor.AsyncioSelectorReactor"
+        )
+
+        self.loop = set_asyncio_event_loop(None)
+
+        self.connector = aiohttp.TCPConnector(
+            limit_per_host=settings.getint("CONCURRENT_REQUESTS_PER_DOMAIN"),
+            loop=self.loop,
+        )
+        self.session = aiohttp.ClientSession(connector=self.connector, loop=self.loop)
+        self._crawler = crawler
+
+    @classmethod
+    def from_crawler(cls, crawler: Crawler) -> Self:
+        return cls(crawler.settings, crawler)
+
+    def download_request(self, request: Request, spider: Spider) -> Deferred[Response]:
+        return deferred_from_coro(self._download_request(request))
+
+    async def _download_request(self, request: Request):
+        """download through aiohttp interface"""
+
+        proxy = request.meta.get("proxy")
+        timeout = aiohttp.ClientTimeout(total=request.meta.get("download_timeout"))
+
+        url = request.url
+        method = request.method
+        body = request.body
+        headers = None if request.headers is None else request.headers.to_unicode_dict()
+        if isinstance(request.cookies, dict):
+            cookies = request.cookies
+        else:
+            cookies = {
+                str(item["name"]): str(item["value"]) for item in request.cookies
+            }
+        body = request.body
+
+        self.session.cookie_jar.update_cookies(cookies)
```

Comment:
Hmm I'm not sure what should we do with the builtin aiohttp cookie handling as we want to do that ourselves.